src/assets/AssetsContractController.ts

```diff
@@ -356,9 +390,11 @@ export class AssetsContractController extends BaseController<
     selectedAddress: string,
     tokensToDetect: string[],
   ) {
+    const contractAddress =
+      SINGLE_CALL_BALANCES_ADDRESS_BY_CHAINID[this.config.chainId];
     const contract = this.web3.eth
       .contract(abiSingleCallBalancesContract)
-      .at(SINGLE_CALL_BALANCES_ADDRESS);
+      .at(contractAddress);
```

Comment:
It seems that this change makes the contract address dynamic depending on which network is being used. That said, I would expect to see a test where one network is selected, this method is called, some assertion is made, then the network is changed, this method is called again, and a different assertion is made.

Reply:
Same answer as above - Just added test that calls getBalancesInSingleCall after switching network. Under the hood, it will call this function to update chainId on AssetsContractController

Reply:
That's great, thanks!